pallets/parachain-staking/src/migrations.rs

```diff
@@ -34,9 +40,210 @@ use frame_support::{
+pub struct SplitDelegatorStateIntoDelegationScheduledRequests<T>(PhantomData<T>);
+
+impl<T: Config> SplitDelegatorStateIntoDelegationScheduledRequests<T> {
+	const PALLET_PREFIX: &'static [u8] = b"ParachainStaking";
+	const DELEGATOR_STATE_PREFIX: &'static [u8] = b"DelegatorState";
+
+	#[allow(deprecated)]
+	#[cfg(feature = "try-runtime")]
+	fn old_request_to_string(
+		delegator: &T::AccountId,
+		request: &crate::deprecated::DelegationRequest<T::AccountId, BalanceOf<T>>,
+	) -> String {
+		match request.action {
+			DelegationChange::Revoke => {
+				format!(
+					"delegator({:?})_when({})_Revoke({:?})",
+					delegator, request.when_executable, request.amount
+				)
+			}
+			DelegationChange::Decrease => {
+				format!(
+					"delegator({:?})_when({})_Decrease({:?})",
+					delegator, request.when_executable, request.amount
+				)
+			}
+		}
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn new_request_to_string(request: &ScheduledRequest<T::AccountId, BalanceOf<T>>) -> String {
+		match request.action {
+			DelegationAction::Revoke(v) => {
+				format!(
+					"delegator({:?})_when({})_Revoke({:?})",
+					request.delegator, request.when_executable, v
+				)
+			}
+			DelegationAction::Decrease(v) => {
+				format!(
+					"delegator({:?})_when({})_Decrease({:?})",
+					request.delegator, request.when_executable, v
+				)
+			}
+		}
+	}
+}
+
+#[allow(deprecated)]
+impl<T: Config> OnRuntimeUpgrade for SplitDelegatorStateIntoDelegationScheduledRequests<T> {
+	fn on_runtime_upgrade() -> Weight {
+		use sp_std::collections::btree_map::BTreeMap;
+
+		log::info!(
+			target: "SplitDelegatorStateIntoDelegationScheduledRequests",
+			"running migration for DelegatorState to new version and DelegationScheduledRequests \
+			storage item"
+		);
+
+		let mut reads: Weight = 0;
+		let mut writes: Weight = 0;
+
+		let mut scheduled_requests: BTreeMap<
+			T::AccountId,
+			Vec<ScheduledRequest<T::AccountId, BalanceOf<T>>>,
+		> = BTreeMap::new();
+		<DelegatorState<T>>::translate(
+			|delegator, old_state: OldDelegator<T::AccountId, BalanceOf<T>>| {
+				reads = reads.saturating_add(1);
+				writes = writes.saturating_add(1);
+
+				for (collator, request) in old_state.requests.requests.into_iter() {
+					let action = match request.action {
+						DelegationChange::Revoke => DelegationAction::Revoke(request.amount),
+						DelegationChange::Decrease => DelegationAction::Decrease(request.amount),
+					};
+					let entry = scheduled_requests.entry(collator.clone()).or_default();
+					entry.push(ScheduledRequest {
+						delegator: delegator.clone(),
+						when_executable: request.when_executable,
+						action,
+					});
+				}
+
+				let new_state = Delegator {
+					id: old_state.id,
+					delegations: old_state.delegations,
+					total: old_state.total,
+					less_total: old_state.requests.less_total,
+					status: old_state.status,
+				};
+				Some(new_state)
+			},
+		);
+
+		writes = writes.saturating_add(scheduled_requests.len() as Weight);
```

Comment:
This was a bit jarring for me, "num writes" `as Weight` seems out of place, and I might prefer to see this counted in the following for loop. I'm completely nitpicking here, though, feel free to ignore.

Reply:
I could do that if it helps you in reading - I was just deeply concerned about incurring additional runtime than needed for the migration.
Alternatively can also add a comment.